flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java

```diff
@@ -0,0 +1,221 @@
+                            });
+        }
+        // TODO: Print the column comment until FLINK-18958 is fixed
+        return sb.toString();
+    }
+
+    static String extractFormattedColumns(ResolvedCatalogBaseTable<?> table, String printIndent) {
+        return table.getResolvedSchema().getColumns().stream()
+                .map(column -> String.format("%s%s", printIndent, getColumnString(column)))
+                .collect(Collectors.joining(",\n"));
+    }
+
+    static Optional<String> extractFormattedWatermarkSpecs(
+            ResolvedCatalogBaseTable<?> table, String printIndent) {
+        if (!table.getResolvedSchema().getWatermarkSpecs().isEmpty()) {
+            return Optional.of(
+                    table.getResolvedSchema().getWatermarkSpecs().stream()
+                            .map(
+                                    watermarkSpec ->
+                                            String.format(
+                                                    "%sWATERMARK FOR %s AS %s",
+                                                    printIndent,
+                                                    EncodingUtils.escapeIdentifier(
+                                                            watermarkSpec.getRowtimeAttribute()),
+                                                    watermarkSpec
+                                                            .getWatermarkExpression()
+                                                            .asSerializableString()))
+                            .collect(Collectors.joining("\n")));
+        }
+        return Optional.empty();
+    }
+
+    static Optional<String> extractFormattedComment(ResolvedCatalogBaseTable<?> table) {
+        String comment = table.getComment();
+        if (StringUtils.isNotEmpty(comment)) {
+            return Optional.of(EncodingUtils.escapeSingleQuotes(comment));
+        }
+        return Optional.empty();
+    }
+
+    static Optional<String> extractFormattedPartitionedInfo(ResolvedCatalogTable catalogTable) {
+        if (!catalogTable.isPartitioned()) {
+            return Optional.empty();
+        }
+        return Optional.of(
+                catalogTable.getPartitionKeys().stream()
+                        .map(EncodingUtils::escapeIdentifier)
+                        .collect(Collectors.joining(", ")));
+    }
+
+    static Optional<String> extractFormattedOptions(
+            ResolvedCatalogBaseTable<?> table, String printIndent) {
+        if (Objects.isNull(table.getOptions()) || table.getOptions().isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(
+                table.getOptions().entrySet().stream()
+                        .map(
+                                entry ->
+                                        String.format(
+                                                "%s\'%s\' = \'%s\'",
+                                                printIndent, entry.getKey(), entry.getValue()))
```

Comment:
I don't understand what's going on here. Shouldn't this be this?

```suggestion
                                                "%s'%s' = '%s'",
                                                printIndent, EncodingUtils.escapeSingleQuotes(entry.getKey()), EncodingUtils.escapeSingleQuotes(entry.getValue())))
```